sql/core/src/main/scala/org/apache/spark/sql/execution/command/views.scala

```diff
@@ -131,6 +131,26 @@ case class CreateViewCommand(
         s"specified by CREATE VIEW (num: `${userSpecifiedColumns.length}`).")
     }
 
+    // When creating a permanent view, not allowed to reference temporary objects.
+    if (!isTemporary) {
+      child.collect {
+        // Disallow creating permanent views based on temporary views.
+        case s: UnresolvedRelation
+            if sparkSession.sessionState.catalog.isTemporaryTable(s.tableIdentifier) =>
+          throw new AnalysisException(s"Not allowed to create a permanent view $name by " +
+            s"referencing a temp view ${s.tableIdentifier}. " +
+            originalText.map(sql => s"""SQL: "$sql".""").getOrElse(""))
```

Comment:
do we really need to show the entire sql? it's basically what the user just typed in